### PR TITLE
Adds a new BEPIS design; The Mauna Mug

### PR DESCRIPTION
**Pull request by ghost**
:cl: Kryson & Qustinnus
 add: New BEPIS design; a self heating mug
/:cl:

A self-heating mug that you can get from BEPIS designs. the better the cell thats in it, the higher the max temp and the faster the temp goes up.

### CODE REVIEW

File: code/modules/reagents/reagent_containers/maunamug.dm [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	update_icon()
+	if(reagents.chem_temp >= max_temp)
+		change_power_status(FALSE)
+		audible_message("<span class='notice'>The Mauna Mug lets out a happy beep and turns off!</span>")
+		playsound(src, 'sound/machines/chime.ogg', 50)
+
+/obj/item/reagent_containers/glass/maunamug/Destroy()
+	. = ..()
+	if(cell)
+		qdel(cell)
```
Comment (nemvar) [inline code/modules/reagents/reagent_containers/maunamug.dm]: QDEL_NULL

File: code/modules/reagents/reagent_containers/maunamug.dm [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+/obj/item/reagent_containers/glass/maunamug/proc/change_power_status(status)
+	on = status
+	if(on)
+		START_PROCESSING(SSobj, src)
+	else
+		STOP_PROCESSING(SSobj, src)
+	update_icon()
+
+/obj/item/reagent_containers/glass/maunamug/screwdriver_act(mob/living/user, obj/item/I)
+	open = !open
```
Comment (nemvar) [inline code/modules/reagents/reagent_containers/maunamug.dm]: Call parent

File: code/modules/reagents/reagent_containers/maunamug.dm [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	reagents.adjust_thermal_energy(0.8 * cell.maxcharge * reagents.total_volume, max_temp = max_temp) // 4 kelvin every tick on a basic cell. 160k on bluespace
+	reagents.handle_reactions()
+	update_icon()
+	if(reagents.chem_temp >= max_temp)
+		change_power_status(FALSE)
+		audible_message("<span class='notice'>The Mauna Mug lets out a happy beep and turns off!</span>")
+		playsound(src, 'sound/machines/chime.ogg', 50)
+
+/obj/item/reagent_containers/glass/maunamug/Destroy()
+	. = ..()
```
Comment (nemvar) [inline code/modules/reagents/reagent_containers/maunamug.dm]: Put this at the end of the proc.

File: code/modules/power/cell.dm [hunk 4]
```diff
@@ -225,6 +225,7 @@
 	icon_state = "h+cell"
 	maxcharge = 15000
 	chargerate = 2250
+	rating = 2
```
Comment (nemvar) [inline code/modules/power/cell.dm]: Does this cause any side effects?

Comment (Qustinnus) [reply code/modules/power/cell.dm]: It puts them higher in the ratings for the part switching tool thingy

Comment (GunHog) [reply code/modules/power/cell.dm]: Powercell ratings are determined by charge. There is no need to add ratings to the other cell types, as this is already handled.
https://github.com/tgstation/tgstation/blob/aa8e09ed021228c1e77bffeb28d7f033ab53c702/code/modules/power/cell.dm#L161-L162

**Comment by ArcaneMusic** [comment]
Based and Mugpilled. Godspeed.

**Comment by necromanceranne** [comment]
Mugbombs incoming. This will be damn fun.

File: code/modules/power/cell.dm (hunk 4 above)
Comment (Qustinnus) [reply code/modules/power/cell.dm]: yeah but the tiers are important to know what level of tech they are at. This is just the cells using them wrong.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/reagents/reagent_containers/maunamug.dm [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+	var/open = FALSE
+	var/on = FALSE
+
+/obj/item/reagent_containers/glass/maunamug/Initialize(mapload, vol)
+	. = ..()
+	cell = new /obj/item/stock_parts/cell(src)
+
+/obj/item/reagent_containers/glass/maunamug/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>The status display reads: Current temperature: <b>[reagents.chem_temp]K</b> Current Charge:[cell ? "[cell.charge / cell.maxcharge * 100]%" : "No cell found"].</span>"
```
[the remark this answers is not in the document]
Comment (Qustinnus) [reply code/modules/reagents/reagent_containers/maunamug.dm]: I can't find it.

**Comment by cacogen** [comment]
>Q: how can I join the mauna kea community that is protecting the sacred mountain?
>A: PROTECTING THE MOUNTAIN

**Comment by ArcaneMusic** [comment]
ha HA NOW I'M NOT THE ONLY MERGED CONTRIBUTOR TO THE BEPIS

**Pull request merged**